scipy/special/basic.py

```diff
@@ -2229,29 +2229,58 @@ def perm(N, k, exact=False):
         return vals
 
 
+# http://stackoverflow.com/a/16327037/125507
+def _range_prod(lo, hi):
+    """
+    Product of a range of numbers
+
+    Returns the product of
+    lo * (lo+1) * (lo+2) * ... * (hi-2) * (hi-1) * hi
+    = hi! / (lo-1)!
+    """
+    if lo + 1 < hi:
+        mid = (hi + lo) // 2
+        return _range_prod(lo, mid) * _range_prod(mid + 1, hi)
+    if lo == hi:
+        return lo
+    return lo * hi
+
+
 def factorial(n, exact=False):
-    """The factorial function, n! = special.gamma(n+1).
+    """
+    The factorial of a number or array of numbers
```

Comment:
Full stop at the end of the sentence please. Otherwise, the summary is garbled in the rendered docs.